msal/src/main/java/com/microsoft/identity/client/PublicClientApplication.java

```diff
@@ -632,71 +624,99 @@ public void onError(MsalException exception) {
+                        if (application instanceof ISingleAccountPublicClientApplication) {
+                            listener.onCreated((ISingleAccountPublicClientApplication) application);
+                        } else {
+                            if (application.getConfiguration().mAccountMode != AccountMode.SINGLE) {
+                                listener.onError(new MsalClientException("AccountMode in configuration is not set to single"));
+                                return;
+                            }
+                            listener.onError(new MsalClientException("A single account public client application could not be created for unknown reasons"));
+                        }
+                    }
 
-    protected PublicClientApplication(@NonNull final Context context,
-                                      @NonNull final String clientId) {
-        this(context, (PublicClientApplicationConfiguration) null);
-        mPublicClientConfiguration.mClientId = clientId;
-        initializeApplication();
+                    @Override
+                    public void onError(MsalException exception) {
+                        listener.onError(exception);
+                    }
+                }
+        );
     }
+    //endregion
 
-    protected PublicClientApplication(@NonNull final Context context,
-                                      @NonNull final String clientId,
-                                      @NonNull final String authority) {
+    protected PublicClientApplication(@NonNull final PublicClientApplicationConfiguration configFile,
+                                      @Nullable final String clientId,
+                                      @Nullable final String authority) {
 
-        this(context, clientId);
+        mPublicClientConfiguration = configFile;
 
-        mPublicClientConfiguration.getAuthorities().clear();
+        if (clientId != null) {
+            mPublicClientConfiguration.mClientId = clientId;
+        }
 
         if (authority != null) {
+            mPublicClientConfiguration.getAuthorities().clear();
+
             Authority authorityObject = Authority.getAuthorityFromAuthorityUrl(authority);
             authorityObject.setDefault(true);
             mPublicClientConfiguration.getAuthorities().add(authorityObject);
         }
 
-        Authority.addKnownAuthorities(mPublicClientConfiguration.getAuthorities());
+        initializeApplication();
     }
 
     private void initializeApplication() {
         final Context context = mPublicClientConfiguration.getAppContext();
+        setupTelemetry(context, mPublicClientConfiguration);
+
+        AzureActiveDirectory.setEnvironment(mPublicClientConfiguration.getEnvironment());
+        Authority.addKnownAuthorities(mPublicClientConfiguration.getAuthorities());
+
+        if (mPublicClientConfiguration.getOAuth2TokenCache() instanceof MsalOAuth2TokenCache) {
```

Comment:
Not part of this change but can we move this to helper method like `initializeTSL`